### PR TITLE
Add a guide on how to use the Ruby Octokit SDK

### LINKED ISSUE #23844: Add a guide on how to use the Ruby Octokit SDK
The GitHub docs now include a guide about how to use GitHub's JavaScript Octokit SDK to use the REST API: https://docs.github.com/rest/guides/scripting-with-the-rest-api-and-javascript

We would like to add a similar guide for the [Ruby Octokit SDK](https://github.com/octokit/octokit.rb).

The guide should:

- Follow the same format as https://docs.github.com/rest/guides/scripting-with-the-rest-api-and-javascript. You can copy this guide to use as a template, but please be sure to update all JavaScript-specific content.
- Follow best practices for Ruby, including adhering to a standard Ruby linter
- Not recommend OAuth Apps, but instead recommend [GitHub Apps](https://docs.github.com/en/apps/creating-github-apps/setting-up-a-github-app/about-creating-github-apps)

### PR DESCRIPTION
**Pull request by Manas-Kenge**
### Why: Added a guide on how to use the Ruby Octokit SDK

Closes: #23844

### What's being changed (if available, include any code snippets, screenshots, or gifs):
Added a file 

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by steveward** [comment]
Thanks for the contribution @Manas-Kenge! I'll get this triaged for review. ⚡

File: content/rest/guides/scripting-with-the-rest-api-and-ruby.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+If your workflow needs to access resources outside of the workflow's repository, then you will not be able to use `GITHUB_TOKEN`. In that case, store your credentials as a secret and replace `GITHUB_TOKEN` in the examples below with the name of your secret. For more information about secrets, see "Encrypted Secrets."
+
+If you use the `run` keyword to execute your Ruby script in your {% data variables.product.prodname_actions %} workflows, you can store the value of `GITHUB_TOKEN` as an environment variable. Your script can access the environment variable as `ENV['VARIABLE_NAME']`.
+
+For example, this workflow step stores `GITHUB_TOKEN` in an environment variable called `TOKEN`:
+
+```yaml
+- name: Run script
+  env:
+    TOKEN: ${{ secrets.`GITHUB_TOKEN` }}
```
Comment (skedwards88) [inline content/rest/guides/scripting-with-the-rest-api-and-ruby.md:83]: This should fix the error that is preventing the preview from generating:
```suggestion
    TOKEN: {% raw %}${{ secrets.`GITHUB_TOKEN` }}{% endraw %}
```

**Review by felicitymay — COMMENTED** [review]
Hi @Manas-Kenge 👋🏻 

Thanks for opening this pull request and responding to our request for help creating this article ✨ 

I've made a couple of commits to your branch which should fix the tests and enable a preview of the new article to be deployed. That will make it easier for someone to review your new article.

_Update: it looks as if it least one of my find and replace checks was flawed. Taking another look._

**Comment by skedwards88** [comment]
Thank you for this PR 🚀 ✨ 

We pushed some changes to get the tests passing and render a preview. You can see a preview of your new article here: https://docs-29083-2e2021.preview.ghdocs.com/en/rest/guides/scripting-with-the-rest-api-and-ruby

At first glance this looks great! I'm going to request a technical review from the SDK team, then we'll have a writer review it.

**Review by nickfloyd — APPROVED** [review]
These changes look great from a SDK perspective.  Thank you for doing them ❤️!

**Review by jhosman — COMMENTED** [review]
@Manas-Kenge 👋 Thanks for opening this PR! Great job on this - I'm going to make a few very minor updates to fix some links and make adjustments for style/consistency. Once that's done and tests are passing, we'll get this merged for you. 

If you're looking for other ways to contribute and are up for a challenge, check out our [pumpkin-spice label](https://github.com/github/docs/issues?q=is%3Aissue+is%3Aopen+label%3A%22pumpkin-spice%22+) to find help wanted issues that are a bit more in-depth.

Thanks again for your contribution! ✨ 🚀

File: content/rest/guides/scripting-with-the-rest-api-and-ruby.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+repo = "docs"
+pull_number = 22809
+comment_url = comment_if_data_files_changed(octokit, owner, repo, pull_number)
+
+puts "A comment was added to the pull request: #{comment_url}"
+```
+
+{% note %}
+**Note**: This is just a basic example. In practice, you may want to use error handling and conditional checks to handle various scenarios.
+{% endnote %}
```
Comment (jhosman) [inline content/rest/guides/scripting-with-the-rest-api-and-ruby.md:325]:
```suggestion
{% note %}

**Note:** This is just a basic example. In practice, you may want to use error handling and conditional checks to handle various scenarios.

{% endnote %}
```

**Review by jhosman — APPROVED** [review]

**Pull request merged**